src/ptype2-dispatch.c

```diff
@@ -46,24 +46,107 @@ SEXP vec_ptype2_dispatch(SEXP x, SEXP y,
 static SEXP fns_vec_ptype2_dispatch_s3 = NULL;
 static SEXP syms_vec_ptype2_dispatch_s3 = NULL;
 
+static inline SEXP vec_ptype2_default(SEXP x,
+                                      SEXP y,
+                                      SEXP x_arg,
+                                      SEXP y_arg) {
+  return vctrs_eval_mask4(Rf_install("vec_default_ptype2"),
+                          syms_x, x,
+                          syms_y, y,
+                          syms_x_arg, x_arg,
+                          syms_y_arg, y_arg,
+                          vctrs_ns_env);
+}
+
+static SEXP get_ptype2_method(SEXP x,
+                              const char* generic,
+                              SEXP table,
+                              SEXP* method_sym_out) {
+  SEXP class = R_NilValue;
+  if (OBJECT(x)) {
+    class = Rf_getAttrib(x, R_ClassSymbol);
+  }
+
+  // This also handles gremlins objects where `x` is an OBJECT(), but
+  // the class is NULL
```

Comment:
```suggestion
  // Handle gremlin objects where `x` is an OBJECT(), but class is NULL
```

Reply:
I'll be more explicit that the other case handled here is bare objects.